### DISCUSSION TITLE
60 Jupyter notebook examples for using MapLibre with Leafmap

### DISCUSSION THREAD

**Discussion by giswqs** [maplibre/maplibre-gl-js#4315] [Show and tell]
Check out 60 notebook examples for using MapLibre with [Leafmap](https://github.com/opengeos/leafmap/), which builds upon the MapLibre Python bindings - [py-maplibregl](https://github.com/eodaGmbH/py-maplibregl). 57 out of the 99 [MapLibre JS examples](https://maplibre.org/maplibre-gl-js/docs/examples/) are now also available in Python. 

See below for some examples. 
https://leafmap.org/maplibre/overview/

## Display buildings in 3D

Use extrusions to display buildings' height in 3D.

[![](https://i.imgur.com/9QeicaE.png)](https://leafmap.org/maplibre/3d_buildings)

## Extrude polygons for 3D indoor mapping

Create a 3D indoor map with the fill-extrude-height paint property.

[![](https://i.imgur.com/eYhSWaT.png)](https://leafmap.org/maplibre/3d_indoor_mapping)

## 3D Terrain

Go beyond hillshade and show elevation in actual 3D.

[![](https://i.imgur.com/sjXZ2Jm.jpeg)](https://leafmap.org/maplibre/3d_terrain)

## Add a default marker

Add a default marker to the map.

[![](https://i.imgur.com/ufmqTzx.png)](https://leafmap.org/maplibre/add_a_marker)

## Add deck.gl layers

Add deck.gl layers to the map.

[![](https://i.imgur.com/rQR4687.png)](https://leafmap.org/maplibre/add_deckgl_layer)

## Add an icon to the map

Add an icon to the map from an external URL and use it in a symbol layer.

[![](https://i.imgur.com/Nq1uV9d.png)](https://leafmap.org/maplibre/add_image)

## Add a generated icon to the map

Add an icon to the map that was generated at runtime.

[![](https://i.imgur.com/qWWlnAm.png)](https://leafmap.org/maplibre/add_image_generated)

## Animate a line

Animate a line by updating a GeoJSON source on each frame.

[![](https://i.imgur.com/LRwfBl9.png)](https://leafmap.org/maplibre/animate_a_line)

**Comment by HarelM** [comment]
I would advise to open a PR for awesome maplibre if this is not listed there already.

**Reply by giswqs** [reply]
@HarelM Thanks for the suggestion. PR submitted. 
https://github.com/maplibre/awesome-maplibre/pull/47
